src/libraries/System.Private.CoreLib/src/System/SecureAppContext.cs

```diff
@@ -0,0 +1,64 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+namespace System
+{
+    /// <summary>
+    /// Represents AppContext switches that were in effect when the AppDomain was created.
+    /// </summary>
+    /// <remarks>
+    /// This class provides a place to store the values of security-sensitive switches as they
+    /// were when the application started. Attackers cannot use standard "open reflection"
+    /// gadgets to modify these fields since the reflection stack forbids altering the contents
+    /// of static initonly fields. This provides an extra layer of defense for applications
+    /// which rely on these switches as part of an overall attack surface reduction strategy.
+    ///
+    /// This is not meant to be a perfect defense. A caller can always use unsafe code to modify
+    /// these static fields. However, we assume such a caller is already running code within the
+    /// process. Arbitrary memory writes can also alter these fields. Both of these scenarios are
+    /// outside the scope of our threat model.
+    /// </remarks>
+    internal static class SecureAppContext
```

Comment:
I would make this local to System.Runtime.Serialization.Formatters. I do not think it need to live in CoreLib.

Reply:
I disagree, this has the basis of becoming a wider used feature for disabling things, and isn't serialization specific.

Reply:
If you want this to be a wider used feature for disabling things, this needs to be designed a proper public API/surface. The current shape is not suitable as public API.

Reply:
At least some part of this needs to live in System.Private.CoreLib, as it's required to be able to trim away the reflection logic in _ResourceReader.Core.cs_.